keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -0,0 +1,2495 @@
+| label selectors                          | WH                           | phase 1                                |
+| resource apiGroup + resource             | WH/Audit/P&F/RBAC            | phase 1                                |
+| apiVersion                               | WH                           | phase 1                                |
+| resource name                            | Audit/RBAC                   | phase 1                                |
+| scope (cluster\|namespace)               | WH/P&F                       | phase 1                                |
+| operation (HTTP verb)                    | WH/Audit/P&F                 | phase 1                                |
+| exclude                                  | Audit (level=None)           | phase 1, see below "Exclude matching"  |
+| apiVersion + kind                        |                              | TBD                                    |
+| NonResourceURLs                          | Audit/RBAC/P&F               | No                                     |
+| default / fallthrough matching           |                              | phase 2 see "Default matching" section |
+| user/userGroup                           | Audit                        | phase 2 see "Secondary Authz" section  |
+| permissions (RBAC verb)                  | RBAC                         | phase 2 see "Secondary Authz" section  |
+
+WH = Admission webhooks, P&F = Priority and Fairness
+
+Match criteria must be declared in the `spec.match` field of policy
+configuration resources (see `ReplicaLimit` in the above example) and will be
+declared with API types in a format similar to admission webhooks, P&F, RBAC and
+Audit. Match criteria will also use ordered list of rules similar to these other
+systems.
+
+In order for policy configuration resources to declare match criteria, the
+corresponding configuration CRD schema must have a `spec.match` property. This
+property must conform to the below "matching schema template". This ensures that
+the match criteria is in the format that API server expects (the API server will
+be using duck typing here since there is no established way to do polymorphism
+across CRDs). The schemas of these fields in the configuration CRD may omit any
+optional properties; policy definition authors should only include the parts of
+the "match schema template" that are useful for configuring a particular policy.
+
+Proposed matching mechanism:
+
+- Ordered list of matches. First match that is true determine if the request is
+  validated by the policy-- if exclude is false, it is validated, otherwise it
+  is not.
+- Supported criteria for each match:
+  - GVR (resource apiGroup + resource)
+  - scope (cluster|namespace) 
+  - operation (HTTP verb) 
+  - exclude (if criteria matches, match result is "no match")
+  - namespace selectors
+  - object selectors
+  - resource name
+  - user/userGroup
```

Comment:
user.Extra?  (pod claims are here)

also, this is as far as I got.

Reply:
Added.